cpmpy/transformations/normalize.py

```diff
@@ -34,3 +37,118 @@ def unravel(lst, append):
     unravel((cpm_expr,), append)
 
     return newlist
+
+
+def simplify_boolean(lst_of_expr, num_context=False):
+    """
+    removes boolean constants from all CPMpy expressions
+    only resulting boolean constant is literal 'false'
+    - list_of_expr: list of CPMpy expressions
+    """
+
+    newlist = []
+    for expr in lst_of_expr:
+
+        if isinstance(expr, bool):
+            # not sure if this should happen here or at construction time
+            expr = BoolVal(expr)
```

Comment:
there can be Py bools here yes.

you can check the num_context here and add to newlist directly, instead of relying on the next if to do that then (which can then become an elif)